### ISSUE TITLE
Do not create an empty `stats_file`

### ISSUE THREAD

**Issue by benoit74** [openzim/zimit#211]
`zimit` creates an empty `stats_file` (see [here](https://github.com/openzim/zimit/blob/12dab25e61b2abde9b6c7778a1441680e6a4daf9/zimit.py#L44)) which in turn causes issues on Zimfarm "failed to load progress details: Expecting value: line 1 column 1 (char 0)"

There is probably no need to create this empty `stats_file` since:
- this file is not watched by `inotify`
- this file is a `json_dump`, so overwriten every time we export stats

**Comment by kelson42** [comment]
@benoit74 @rgaudin This ticket raises questions:
* Why an empty stat file is no handled properly by the zimfarm (doing nothing and complaining about non valid stat file)?
* why the stat file is empty in a first place?

**Comment by benoit74** [comment]
> Why an empty stat file is no handled properly by the zimfarm (doing nothing and complaining about non valid stat file)?

Because zimfarm handles properly the case where the file is missing, but not the case where it is empty. We could handle it as well (silence warnings) but since the file content is supposed to be JSON, I consider that it is better to warn when a scrapper is creating an empty file (because it could be linked to other issues).

>     why its not watched by inotify?
inotify is running inside `zimit` to watch crawler + warc2zim log files (specific to this scraper) and transform them into  stats file (common to all zimfarm scraper). Stat file hence does not have to be watched after by the scraper, it is watched by zimfarm worker.

>     why the stat file is empty in a first place?
Because as stated above, inotify has to watch crawler + warc2zim log files ; inotify can only watch existing files ; instead of waiting for crawler / warc2zim to create those files and only after that setup inotify, it has been decided to create those files empty and setting up inotify immediately at the start of the scraper ; most probably by mistake (but this has to be confirmed) the stats file has also been created empty, just because it was already done for the two other files passed to the function and we did not realized at that time it was unneeded / problematic to do it for the stats file

**Comment by rgaudin** [comment]
This `failed to load progress details: Expecting value: line 1 column 1 (char 0)` is a new issue we're seeing with recent zimit. It has never been a problem in the past and I suspect it's due to a change in the crawler's handling of the stats file.

It was created empty on purpose: as you said, we need it to watch it and so the first time the crawler updates it, it notifies and the rest happens.

Since this is a _regression_, we must first understand what caused the change in behavior.

**Comment by benoit74** [comment]
See https://github.com/openzim/zimit/issues/213 to understand why this underlying issue was not discovered before.
